### PR TITLE
docs: Document overriding capabilities and extending settings

### LINKED ISSUE #6283: Document how to override plugin settings/capabilities locally
https://github.com/meltano/hub/issues/580 is the relevant Hub issue to add some text on the specific plugin pages. 

This capability was added in https://github.com/meltano/meltano/pull/6033/files

Likely on this page we should talk about how capabilities can be defined and overriden. https://docs.meltano.com/guide/configuration

A changelog entry for this feature should be added as it was never documented due to the 2.0 rush.

This can be coupled a bit with https://github.com/meltano/meltano/issues/6107 as well.

### PR DESCRIPTION
**Pull request by edgarrmondragon**
Related:

- Closes #6283
- Implementation PR: https://github.com/meltano/meltano/pull/6033

### CODE REVIEW

**Comment by edgarrmondragon** [comment]
Docs preview: https://deploy-preview-6411--meltano.netlify.app/guide/configuration#overriding-discoverable-plugin-properties

**Review by tayloramurphy — CHANGES_REQUESTED** [review]

File: docs/src/_guide/configuration.md [hunk 1]
```diff
@@ -44,6 +44,26 @@ control. Sensitive values like passwords and tokens are most appropriately store
 
 [`meltano config <plugin> set`](/reference/command-line-interface#config) will automatically store configuration in `meltano.yml` or `.env` as appropriate.
 
+### Overriding discoverable plugin properties
+
+Starting with Meltano [`2.0`](/reference/v2-migration), you can override the properties of discoverable plugins and extend the default [`settings`](/reference/settings):
```
[the remark this answers is not in the document]
Comment (edgarrmondragon) [reply docs/src/_guide/configuration.md:49]: > is there anything else besides capabilities and settings that can be overridden?

@tayloramurphy `settings_group_validation` too. Is that worth documenting explicitly too?

[the remark this answers is not in the document]
Comment (tayloramurphy) [reply docs/src/_guide/configuration.md:49]: @edgarrmondragon I think it's worth it. 

Question on the behavior, `capabilities` are fully overridden such that it's a full array replacement, right?

For settings though, does it always append or is it actually comparing each setting and doing a replacement if there's a match? Specific example, if the plugin defines a setting as having type string but the users overrides the type for that setting with type: integer, will there be duplicate settings or is it a smarter merge?

[the remark this answers is not in the document]
Comment (edgarrmondragon) [reply docs/src/_guide/configuration.md:49]: > `capabilities` are fully overridden such that it's a full array replacement, right?

@tayloramurphy correct, they're fully replaced, not extended.

> For settings though, does it always append or is it actually comparing each setting and doing a replacement if there's a match? Specific example, if the plugin defines a setting as having type string but the users overrides the type for that setting with type: integer, will there be duplicate settings or is it a smarter merge?

If there's a collision on name, then the setting, say `token`, is taken from the override definition in `meltano.yml` and used at runtime, while the `token` setting definition in the lockfile is discarded. So no, there aren't any duplicates 🙂.

[the remark this answers is not in the document]
Comment (tayloramurphy) [reply docs/src/_guide/configuration.md:49]: @edgarrmondragon all of that context is worth adding then please 😄

File: docs/src/_guide/configuration.md [hunk 2]
```diff
@@ -44,6 +44,28 @@ control. Sensitive values like passwords and tokens are most appropriately store
...
+    - discover
+    - catalog
+    settings:  # These will be appended to the settings declared in the lockfile
+    - name: my-new-setting
+      kind: object
+      value:
+        key: value
+```
+
+All overrides replace the values of in the [lockfile](/concepts/plugins#lock-artifacts), except for `settings`, which extend the base definitions. If there is a collision on name, then the setting is taken from the override definition in `meltano.yml` and used at runtime, while the token setting definition in the lockfile is discarded.
```
Comment (tayloramurphy) [inline docs/src/_guide/configuration.md:67]:
```suggestion
All overrides replace the values stored in the [lockfile](/concepts/plugins#lock-artifacts), except for `settings`, which extend the base definitions. If there is a collision on name, then the setting is taken from the override definition in `meltano.yml` and used at runtime, while the token setting definition in the lockfile is discarded.
```

**Review by tayloramurphy — APPROVED** [review]

**Pull request merged**